packages/theming/apple-theme/src/appleTypography.ios.ts

```diff
@@ -50,90 +60,18 @@ export function appleTypography(): Typography {
-      },
-      caption1: {
-        face: fontTokens.family.base,
-        size: fontTokens.size[200],
-        weight: fontTokens.weight.regular,
-        lineHeight: fontTokens.lineHeight[200],
-        letterSpacing: -0.08,
-      },
-      caption1Strong: {
-        face: fontTokens.family.base,
-        size: fontTokens.size[200],
-        weight: fontTokens.weight.semibold,
-        lineHeight: fontTokens.lineHeight[200],
-        letterSpacing: -0.08,
-      },
-      body2: {
-        face: fontTokens.family.base,
-        size: fontTokens.size[300],
-        weight: fontTokens.weight.regular,
-        lineHeight: fontTokens.lineHeight[300],
-        letterSpacing: -0.23,
-      },
-      body2Strong: {
-        face: fontTokens.family.base,
-        size: fontTokens.size[300],
-        weight: fontTokens.weight.semibold,
-        lineHeight: fontTokens.lineHeight[300],
-        letterSpacing: -0.23,
-      },
-      body1: {
-        face: fontTokens.family.base,
-        size: fontTokens.size[400],
-        weight: fontTokens.weight.regular,
-        lineHeight: fontTokens.lineHeight[400],
-        letterSpacing: -0.43,
-      },
-      body1Strong: {
-        face: fontTokens.family.base,
-        size: fontTokens.size[400],
-        weight: fontTokens.weight.semibold,
-        lineHeight: fontTokens.lineHeight[400],
-        letterSpacing: -0.43,
-      },
-      title3: {
-        face: fontTokens.family.base,
-        size: fontTokens.size[500],
-        weight: fontTokens.weight.semibold,
-        lineHeight: fontTokens.lineHeight[500],
-        letterSpacing: -0.45,
-      },
-      title2: {
-        face: fontTokens.family.base,
-        size: fontTokens.size[600],
-        weight: fontTokens.weight.semibold,
-        lineHeight: fontTokens.lineHeight[600],
-        letterSpacing: -0.26,
-      },
-      title1: {
-        face: fontTokens.family.base,
-        size: fontTokens.size[700],
-        weight: fontTokens.weight.bold,
-        lineHeight: fontTokens.lineHeight[700],
-        letterSpacing: 0.38,
-      },
-      largeTitle: {
-        face: fontTokens.family.base,
-        size: fontTokens.size[800],
-        weight: fontTokens.weight.bold,
-        lineHeight: fontTokens.lineHeight[800],
-        letterSpacing: 0.4,
-      },
-      display: {
-        face: fontTokens.family.base,
-        size: fontTokens.size[900],
-        weight: fontTokens.weight.bold,
-        lineHeight: fontTokens.lineHeight[900],
-        letterSpacing: 0.26,
-      },
+      caption2: convertAliasFont(aliasTokens.caption2 as FontStyleTokens),
+      caption1: convertAliasFont(aliasTokens.caption1 as FontStyleTokens),
+      caption1Strong: convertAliasFont(aliasTokens.caption1Strong as FontStyleTokens),
+      body2: convertAliasFont(aliasTokens.body2 as FontStyleTokens),
+      body2Strong: convertAliasFont(aliasTokens.body2Strong as FontStyleTokens),
+      body1: convertAliasFont(aliasTokens.body1 as FontStyleTokens),
```

Comment:
Ah, I didn't know this existed. I'll see if we can share any of this logic.

> I'm not quite understanding this:
> 
> > One other thing we needed to do is to add normal and bold as potential FontWeights properties since [these are legal font weight values in React Native](https://reactnative.dev/docs/text-style-props#fontweight).
> 
> Wouldn't that mean that if someone passes in normal or bold into a Text component core-RN should support it? I'm not understanding why that means FURN theming, which is to support Fluent Design, should support it

Our alias font tokens come in the form of [`FontStyleTokens`](https://github.com/microsoft/fluentui-react-native/blob/e07c6548be5d6e18613e46e309ab667040455369/packages/utils/tokens/src/text-tokens.ts#L11-L17). Its `fontWeight` property is allowed to be a key of [`TextStyle.fontWeight`](https://reactnative.dev/docs/text-style-props#fontweight), which includes values of `'normal'` and `'bold'`.

In order to properly convert this to a [`VariantValue`](https://github.com/microsoft/fluentui-react-native/blob/main/packages/theming/theme-types/src/Typography.types.ts#L107-L113)'s [`FontWeight`](https://github.com/microsoft/fluentui-react-native/blob/e07c6548be5d6e18613e46e309ab667040455369/packages/theming/theme-types/src/Typography.types.ts#L92), we must do one of the following:
* Add it as a key of [`FontWeights`](https://github.com/microsoft/fluentui-react-native/blob/e07c6548be5d6e18613e46e309ab667040455369/packages/theming/theme-types/src/Typography.types.ts#L84-L87). This is what I opted for because it seemed to fit well with other other existing names of `regular` and `semibold`.
* Add it to the [`FontWeightValue`](https://github.com/microsoft/fluentui-react-native/blob/e07c6548be5d6e18613e46e309ab667040455369/packages/theming/theme-types/src/Typography.types.ts#L73) enum. This didn't make as much sense since this enum is otherwise full of numbers.
* Add a function that converts these new keywords to their numerical counterparts (that is, `'400'` for `'normal'` and `'700'` for `'bold'`). This seems reasonable as these numerical mappings appear to be standard, but it may not be future-proof if React Native chooses to support new names, which is why I shied away from this option (although the option I went with would require updating things anyway, so maybe it wouldn't make a difference).

Reply:
It doesn't look like there's an elegant way to combine the logic, but I can at least move the relevant parts of code to this package.